application/src/main/java/org/togetherjava/tjbot/commands/moderation/BanCommand.java

```diff
@@ -0,0 +1,181 @@
+import net.dv8tion.jda.api.exceptions.ErrorResponseException;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import net.dv8tion.jda.api.requests.ErrorResponse;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import org.togetherjava.tjbot.config.Config;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * This command can ban users and optionally remove their messages from the past days. Banning can
+ * also be paired with a ban reason. The command will also try to DM the user to inform them about
+ * the action and the reason.
+ * <p>
+ * The command fails if the user triggering it is lacking permissions to either ban other users or
+ * to ban the specific given user (for example a moderator attempting to ban an admin).
+ */
+public final class BanCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(BanCommand.class);
+    private static final String TARGET_OPTION = "user";
+    private static final String DELETE_HISTORY_OPTION = "delete-history";
+    private static final String REASON_OPTION = "reason";
+    private static final String COMMAND_NAME = "ban";
+    private static final String ACTION_VERB = "ban";
+    private final Predicate<String> hasRequiredRole;
+
+    /**
+     * Constructs an instance.
+     */
+    public BanCommand() {
+        super(COMMAND_NAME, "Bans the given user from the server", SlashCommandVisibility.GUILD);
+
+        getData().addOption(OptionType.USER, TARGET_OPTION, "The user who you want to ban", true)
+            .addOption(OptionType.STRING, REASON_OPTION, "Why the user should be banned", true)
+            .addOptions(new OptionData(OptionType.INTEGER, DELETE_HISTORY_OPTION,
+                    "the amount of days of the message history to delete, none means no messages are deleted.",
+                    true).addChoice("none", 0).addChoice("recent", 1).addChoice("all", 7));
+
+        hasRequiredRole = Pattern.compile(Config.getInstance().getHeavyModerationRolePattern())
+            .asMatchPredicate();
+    }
+
+    private static RestAction<InteractionHook> handleAlreadyBanned(@NotNull Guild.Ban ban,
+            @NotNull Interaction event) {
+        String reason = ban.getReason();
+        String reasonText =
+                reason == null || reason.isBlank() ? "" : " (reason: %s)".formatted(reason);
```

Comment:
Yeah I don't like ternary operators..
Things like this make the code too unreadable imo.
Ternary operators should be simple like 
```java
String reasonText = (reason == null) ? "" : reason;
```
This is still readable, but because there's an OR operator in the if, formatting and such it becomes too unreadable imo.

Switch to a if statement?

Reply:
This code is not part of this PR. Closing.